src/libraries/Common/src/System/Security/Cryptography/ECDsaCng.SignVerify.cs

```diff
@@ -20,15 +20,7 @@ public override byte[] SignHash(byte[] hash)
         {
             ArgumentNullException.ThrowIfNull(hash);
 
-            int estimatedSize = KeySize switch
-            {
-                256 => 64,
-                384 => 96,
-                521 => 132,
-                // If we got here, the range of legal key sizes for ECDsaCng was expanded and someone didn't update this switch.
-                // Since it isn't a fatal error to miscalculate the estimatedSize, don't throw an exception. Just truck along.
-                _ => KeySize / 4,
-            };
+            int estimatedSize = GetMaxSignatureSize(DSASignatureFormat.IeeeP1363FixedFieldConcatenation);
```

Comment:
Is this more expensive? The implementation of the method isn't trivial, but maybe it doesn't matter?

Reply:
Maybe in a micro-benchmark, but ECDSA signing and verifying is going to dominate. For `IeeeP1363FixedFieldConcatenation` it basically boils down to `AsymmetricAlgorithmHelpers.BitsToBytes(KeySize) * 2`; Where `BitsToBytes` just does the "ceiling then divide by 8". I wouldn't be surprised if the JIT inlined `BitsToBytes`.